src/Features/LanguageServer/Protocol/Handler/DocumentCache.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler;
+
+internal sealed class DocumentCache : ResolveCache<TextDocumentIdentifier>
+{
+    public DocumentCache() : base(maxCacheSize: 3)
```

Comment:
what does it mean to cache 3 text document identifiers?  when would we use this?  what would it mean to drop something from the cache?